### PR TITLE
Kratos 157 get api for existing sso config

### PR DESCRIPTION
**Pull request by vipin230**
### :nut_and_bolt: Description: What code changed, and why?
Use-case:- If admin user wants to check their sso config configured in automate for saml login.

This GET api can be used by UI team to introduce this feature in UI.

### :+1: Definition of Done

- Create a GET Api for sso config
- Tested api with below scenarios.

**Scenarios tested:-**
1. api-token:- 
      - admin
      - editors
      - viewers
2.  Authorization: Bearer:-
     - admin
     - editors
     - viewer
3. Deployment Type:-
     - Standalone 
     - HA
     - SAAS

### :athletic_shoe: How to Build and Test the Change

1. rebuild automate-gateway and automate-deployment pkgs.
5. create a api token with admin privileges.
6. Make GET api request from postman on `https://{{url}}/api/v0/sso/config`
url - your automate url
 
Demo:- [Get api demo](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EX2spEg8BSlGiBWuAVgmr3YByvJ4kaijouc4KZ5G9KIyyg?e=NY52qj)
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: api/external/sso/sso.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+			get: "/api/v0/sso/config"
+		};
+		option (chef.automate.api.iam.policy) = {
+			resource: "sso:config"
+			action: "sso:config:get"
+		};
+	};
+}
+
+message GetSsoConfigRequest {}
```
[the remark this answers is not in the document]
Comment (vipin230) [reply api/external/sso/sso.proto]: Resolved

File: api/external/sso/sso.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package chef.automate.api.sso;
+option go_package = "github.com/chef/automate/api/external/sso";
+
+import "google/api/annotations.proto";
+import "external/annotations/iam/annotations.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+
+service SsoConfigService {
+    rpc GetSsoConfig (GetSsoConfigRequest) returns (GetSsoConfigResponse) {
```
[the remark this answers is not in the document]
Comment (vipin230) [reply api/external/sso/sso.proto]: Resolved

[the remark this answers is not in the document]
Comment (vipin230) [reply api/external/sso/sso.proto]: Resolved

File: components/automate-gateway/handler/ssoconfig.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import (
+	"context"
+
+	deployment "github.com/chef/automate/api/interservice/deployment"
+
+	"github.com/chef/automate/api/external/sso"
+)
+
+// SsoConfig - the ssoconfig service data structure
+type Server struct {
```
[the remark this answers is not in the document]
Comment (vipin230) [reply components/automate-gateway/handler/ssoconfig.go]: Resolved

File: components/automate-gateway/handler/ssoconfig.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	sso_url := new.SsoUrl.GetValue()
+	email_attr := new.EmailAttr.GetValue()
+	username_attr := new.UsernameAttr.GetValue()
+	entity_issuer := new.EntityIssuer.GetValue()
+	group_attr := new.GroupsAttr.GetValue()
+	allowed_group := new.AllowedGroups
+	name_id_policy_format := new.NameIdPolicyFormat.GetValue()
+
+	return &sso.GetSsoConfigResponse{
+		CaContents:         ca_contents,
```
[the remark this answers is not in the document]
Comment (vipin230) [reply components/automate-gateway/handler/ssoconfig.go]: Resolved

**Review by Dmaddu — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Review by rishabhjhs — APPROVED** [review]

**Review by vviveksharma — APPROVED** [review]

**Review by sandhi18 — APPROVED** [review]

**Pull request merged**